### ISSUE TITLE
Orange Pi 3B shows constant Temperature -1

### ISSUE THREAD

**Issue by mo7390** [klattimer/Malaria#8]
Hi klattimer, thank you for this script it's working well (after debugging paho-mqtt 2.0.0 issue (had to downgrade back to 1.6.1 manually) to get the MQTT to connect.

I'm using an Orange Pi 3B and I'm running Ubuntu 2.04 Jammy LTS and have noticed that the CPU temp shows constantly as -1, I've ensured psutils is installed as I believe this is what the thermal plugin pulls it's data from, and this must be an Orange Pi 3B thing?

Are there any recommendations to modify my install for the OPi3B? 

Love the small footprint and the quantity of useful data, particularly with the HA integration , keep up the good work :)

Matt

**Comment by klattimer** [comment]
Yeah it looks like a few days ago MQTT updated, and I'm waiting for the storm to chill out a bit before I change the code to the new stuff. 

I'm not sure about the orange pi 3B, but you're right that the temperature comes from psutils, so if it's not being read properly there, that would be an upstream issue I think. 

This might help you debug a little. 
https://forum.armbian.com/topic/904-how-to-check-temperature-of-orangepi-pc/

**Comment by klattimer** [comment]
Also, I'm working on HA integration at the moment, so there'll be some changes which might break things as I bounce through GitHub for testing on various devices. Sorry for that, by the end it should be nicer.

**Comment by mo7390** [comment]
Thanks klattimer, psutils works fine when run in python on the board itself, I just can't get it to upload into mqtt but I'll use the above resources you posted to see if I can massage it, more than likely just a paho-mqtt thing so I'm happy to wait :)

**Comment by mo7390** [comment]
python3
>>> import psutil
>>> psutil.sensors_temperatures()
{'soc_thermal': [shwtemp(label='', current=39.444, high=115.0, critical=115.0)],                                                                                                                          'gpu_thermal': [shwtemp(label='', current=38.333, high=None, critical=None)]}
>>>

but when arriving in mosquitto it only shows
![image](https://github.com/klattimer/Malaria/assets/69316954/7b2ae692-f9f1-48e0-8b7e-b965de3bdd0e)

**Comment by klattimer** [comment]
Weird. The bit of code you should look at is here. 

https://github.com/klattimer/Malaria/blob/main/Malaria/Plugins/thermal.py 

There really isn't much to it, so not sure what's going on here.
